src/lib/GovernorTimelockCompound.sol

```diff
@@ -0,0 +1,246 @@
+   * @dev The _timelock.gracePeriod() was changed from _timelock.GRACE_PERIOD() in the original Openzeppelin contract.
+   *
+   * Original Openzeppelin source:
+   * https://github.com/OpenZeppelin/openzeppelin-contracts/blob/49c0e4370d0cc50ea6090709e3835a3091e33ee2/contracts/governance/extensions/GovernorTimelockCompound.sol#L67
+   */
+  function state(uint256 proposalId)
+    public
+    view
+    virtual
+    override(IGovernor, Governor)
+    returns (ProposalState)
+  {
+    ProposalState status = super.state(proposalId);
+
+    if (status != ProposalState.Succeeded) return status;
+
+    uint256 eta = proposalEta(proposalId);
+    if (eta == 0) return status;
+    else if (block.timestamp >= eta + _timelock.gracePeriod()) return ProposalState.Expired;
+    else return ProposalState.Queued;
+  }
+
+  /**
+   * @dev Public accessor to check the address of the timelock
+   */
+  function timelock() public view virtual override returns (address) {
+    return address(_timelock);
+  }
+
+  /**
+   * @dev Public accessor to check the eta of a queued proposal
+   */
+  function proposalEta(uint256 proposalId) public view virtual override returns (uint256) {
+    return _proposalTimelocks[proposalId].timer.getDeadline();
+  }
+
+  /**
+   * @dev Function to queue a proposal to the timelock.
+   */
+  function queue(
+    address[] memory targets,
+    uint256[] memory values,
+    bytes[] memory calldatas,
+    bytes32 descriptionHash
+  ) public virtual override returns (uint256) {
+    uint256 proposalId = hashProposal(targets, values, calldatas, descriptionHash);
+
+    require(state(proposalId) == ProposalState.Succeeded, "Governor: proposal not successful");
+
+    uint256 eta = block.timestamp + _timelock.delay();
+    _proposalTimelocks[proposalId].timer.setDeadline(eta.toUint64());
+    for (uint256 i = 0; i < targets.length; ++i) {
+      require(
+        !_timelock.queuedTransactions(
+          keccak256(abi.encode(targets[i], values[i], "", calldatas[i], eta))
+        ),
+        "GovernorTimelockCompound: identical proposal action already queued"
+      );
+      _timelock.queueTransaction(targets[i], values[i], "", calldatas[i], eta);
+    }
+
+    emit ProposalQueued(proposalId, eta);
+
+    return proposalId;
+  }
+
+  /**
+   * @dev Overridden execute function that run the already queued proposal through the timelock.
+   */
+  function _execute(
+    uint256 proposalId,
+    address[] memory targets,
+    uint256[] memory values,
+    bytes[] memory calldatas,
+    bytes32 /*descriptionHash*/
+  ) internal virtual override {
+    uint256 eta = proposalEta(proposalId);
+    require(eta > 0, "GovernorTimelockCompound: proposal not yet queued");
+    // In the original contract, the _timelock is not casted to an address. Failing to cast it causes a compile error "Explicit type conversion not allowed from "contract IRadworksTimelock" to "address payable".".
```

Comment:
nit: this line is long, but i can appreciate why we aren't running `forge fmt` on the file. maybe just break this somewhere?

Reply:
Done 4c8b4aa5079ff2ca8a0124eeab3624cc5cfce0db